homeassistant/components/camera/nest.py

```diff
@@ -0,0 +1,108 @@
+"""
+Support for Nest Cameras.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/camera.nest/
+"""
+
+import logging
+from datetime import timedelta
+import requests
+from homeassistant.components.camera import (PLATFORM_SCHEMA, Camera)
+import homeassistant.components.nest as nest
+from homeassistant.util.dt import utcnow
+
+
+DEPENDENCIES = ['nest']
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({})
+
+NEST_BRAND = "Nest"
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup a Nest Cam."""
+    if discovery_info is None:
+        return
+    camera_devices = hass.data[nest.DATA_NEST].camera_devices()
+    cameras = [NestCamera(structure, device)
+               for structure, device in camera_devices]
+    add_devices(cameras)
+
+
+class NestCamera(Camera):
+    """Representation of a Nest Camera."""
+
+    def __init__(self, structure, device):
+        """Initialize a Nest Camera."""
+        super(NestCamera, self).__init__()
+        self.structure = structure
+        self.device = device
+
+        # data attributes
+        self._location = None
+        self._name = None
+        self._is_online = None
+        self._is_streaming = None
+        self._is_video_history_enabled = False
+        # default to non-NestAware subscribed, but will be fixed during update
+        self._time_between_snapshots = timedelta(seconds=30)
+        self._last_image = None
+        self._next_snapshot_at = None
+
+        # This is so the initial values are set.
+        # Remove this when update gets called correctly.
+        self.update()
```

Comment:
This is expected behavior. As an alternative you can remove this statement and call `add_devices(cameras, True)` instead. This will trigger HASS to update your camera on init.